watch.go

```diff
@@ -124,18 +126,68 @@ type cliContext interface {
 	Int(name string) int
 }
 
+const (
+	rangeDurationParseErrorSubString = "time: invalid duration "
+)
+
+// parseHorizon parses a duration string from the command line In accordance
+// with the most common reason for parse failure (specifying a large number), On
+// an error that looks like a range to large issue, we coerce to the maximum
+// hours and ignore the error. Errors that don't contain the marker substring
+// are returned as is.
+func parseHorizon(horizon string) (time.Duration, error) {
+	d, err := time.ParseDuration(horizon)
+	if err == nil {
+
+		// clamp the horizon, otherwise it may overflow the idtimestamp epoch
+		if d > maxHorizon {
+			return maxHorizon, nil
```

Comment:
Should we let the user know that we have clamped the horizon to a fixed maximum?

Reply:
Yes, ideally. Just an info message would
Be fine

Reply:
Updated to log if we clamp. INFO level